docs/INSTALL.md

```diff
@@ -1,91 +1,69 @@
-To install `pandoc`, visit https://pandoc.org/installing.html and follow instructions
-for Mac OS and Ubuntu.
-
-AWS Textract relies on the following packages for viewing image results: 
-
-https://pypi.org/project/pdf2image/
-
-To install these packages run:
-
-```
-brew install poppler
-```
-
-_If you have omitted SOPN and are having problems getting the project to run, you may need to follow the SOPN steps._
\ No newline at end of file
+## Local development
+
+To develop YNR on your local machine you'll first need to install its
+containerisation prerequisites. We use containers in development to isolate the
+(non-trivial!) set of *application* prerequisites away from your local machine,
+and to get closer to the intended future state of the application's
+*production* deployment.
+
+### Install and test containerisation prerequisites
+
+1. Clone this repository:
+   `git clone --branch jcm/wip https://github.com/DemocracyClub/yournextrepresentative`
+1. Install the `podman` command: https://podman.io/docs/installation.
+   These installation mechanisms have been tested:
+   - System package on Ubuntu 24.04 LTS
+     - https://podman.io/docs/installation#ubuntu
+1. Install `podman-compose` v1.2.0: https://pypi.org/project/podman-compose/.
+   These installation mechanisms have been tested:
+   - Local `pip` installation of v1.2.0 on Ubuntu 24.04 LTS
+     - https://pypi.org/project/podman-compose/
+     - `pip install podman-compose`
+     - Either inside a venv, or not, as you prefer
+   - Manual installation of v1.2.0 APT package on Ubuntu 24.04 LTS
+     - https://packages.ubuntu.com/oracular/all/podman-compose/download
+     - `dkpkg -i path/to/debian-package.deb`
+1. Configure `podman` to be less chatty, by placing this configuration in `$HOME/.config/containers/containers.conf`:
+   ```ini
+   # Don't emit logs on each invocation of the compose command indicating
+   # that an external compose provider is being executed.
+   [engine]
+   compose_warning_logs=false
+   ```
+1. Make sure the `bash` shell is available:
+   `which bash || echo Not found`
+1. Build any container images used by the compose stack:
+   `podman compose build`
+1. Pull any 3rd-party container images used by the compose stack:
+   `podman compose pull`
+1. Set up your development envvars as needed, by placing keys and values in
+   `env/frontend.env`, using `env/frontend.env.example` as a template.
+    In general, the only envar you should need is this:
+   ```
+   DJANGO_SETTINGS_MODULE=ynr.settings
+   ```
+1. Copy `ynr/settings/local.py.container.example` to `ynr/settings/local.py`.
+   If you already have a `ynr/settings/local.py` file, incorporate the example
+   file's settings. **If you don't use most of the example file's settings, you
+   *will* experience problems interacting with the app, later**.
+1. Test that the compose stack can be stood up:
+   ```bash
+   podman compose up -d # NB Space between "podman" and "compose"!
+   curl 0:8080
```

Comment:
FYI, I initially installed `podman-conpose>=1.2.0` and got `1.3.0`. `podman compose up -d` with that version _does not_ detach the process(!), so the first command just hangs around. If I `curl` in another shell, I see a 200 from the service. 

Installing `podman-compose==1.2.0` works as expected.

Reply:
Sorry, it gives a 500, not 200, but it gives the result I expected :sunglasses:

Reply:
Yeah I also pinned to `1.2.0`
https://github.com/DemocracyClub/yournextrepresentative/pull/2485#discussion_r1908475072

Reply:
Would you like a `dev-requirements.txt` to exist somewhere in the repo? My feeling is that we don't need to do that, as it's only coincidence that `podman-compose` even /is/ a pip-installable thing, and there's no strong requirement on a dev to install it using pip rather than (say) apt.

Reply:
Resolved, given @symroe's reply here: https://github.com/DemocracyClub/yournextrepresentative/pull/2485#discussion_r1918247719.